x/stakeibc/keeper/unbonding_records.go

```diff
@@ -86,14 +90,84 @@ func (k Keeper) GetTotalUnbondAmountAndRecordsIds(ctx sdk.Context, chainId strin
 		k.Logger(ctx).Info(utils.LogWithHostZone(chainId, "Epoch %d - Status: %s, Amount: %v",
 			epochUnbonding.EpochNumber, hostZoneRecord.Status, hostZoneRecord.NativeTokenAmount))
 
-		// We'll unbond all records that have status UNBONDING_QUEUE and have an amount g.t. zero
-		if hostZoneRecord.Status == recordstypes.HostZoneUnbonding_UNBONDING_QUEUE && hostZoneRecord.NativeTokenAmount.GT(sdkmath.ZeroInt()) {
-			totalUnbonded = totalUnbonded.Add(hostZoneRecord.NativeTokenAmount)
-			unbondingRecordIds = append(unbondingRecordIds, epochUnbonding.EpochNumber)
-			k.Logger(ctx).Info(utils.LogWithHostZone(chainId, "  %v%s included in total unbonding", hostZoneRecord.NativeTokenAmount, hostZoneRecord.Denom))
+		if hostZoneRecord.ShouldInitiateUnbonding() {
+			epochNumbers = append(epochNumbers, epochUnbonding.EpochNumber)
+			hostZoneUnbondingMap[epochUnbonding.EpochNumber] = *hostZoneRecord
 		}
 	}
-	return totalUnbonded, unbondingRecordIds
+	return epochNumbers, hostZoneUnbondingMap
```

Comment:
small nit, I think `epochNumbers` are just the keys of `hostZoneUnbondingsMap`

would it make sense to just return `hostZoneUnbondingsMap`?

Reply:
I did that originally, but we eventually need them as a list for the callback data. So it seemed a bit cleaner to do the above, rather than have to do something like the following later:
```go
epochNumbers := []uint64{}
for epochNumber, _ := range hostZoneUnbondingMap {
  epochNumbers = append(epochNumbers, epochNumber)
}
```

If only there was a .keys like function haha

Reply:
ah yeah that makes sense!